src/systems/filecoin_markets/storage_market/_index.md

```diff
@@ -3,17 +3,115 @@ menuTitle: Storage Market
+
+A `StorageAsk` contains all the properties that a client will need to determine if a given provider will meet its needs for storage at this moment. Providers should update their asks frequently to insure the information they are providing to clients is up to date.
+
+# Negotiation
+
+Negotiation is how a storage client and a storage provider come to an agreement about a storage deal and reach the point where a deal is published on chain.
+
+Negotiation begins once a client has discovered a miner whose `StorageAsk` meets their desired criteria.The *recommended* order of operations for negotiating and publishing a deal is as follows:
+
+1. Before sending a proposal to the provider, the `StorageClient` adds funds for a deal, as neccesary, to the `StorageMarketActor` (by calling `AddBalance`)
+2. In order to propose a storage deal, the `StorageClient` then calculates the piece commitment (`CommP`) for the data it intends to store ahead of time. This is neccesary so that the `StorageProvider` can verify the data the `StorageClient` sends to be stored matches the `CommP` in the `StorageDealProposal`. For more detail about the relationship between payloads, pieces, and `CommP` see {{<sref piece>}} and {{<sref filproofs>}}
+3. The `StorageClient` now creates a `StorageDealProposal` and sends the proposal and the CID for the root of the data to be stored to the `StorageProvider` using the `Storage Deal Protocol`
+
+Execution now moves to the `StorageProvider`
+
+4. The `StorageProvider` inspects the deal to verify that its parameters match it's own internal criteria (such as price, piece size, deal duration, etc) -- and rejects the proposal if it doesn't
+5. The `StorageProvider` queries the `StorageMarketActor` to verify the `StorageClient` has deposited enough funds to make the deal (i.e. balance > total storage price) and rejects the proposal if it hasn't.
+6. If all criteria are met, the `StorageProvider` responds on the `Storage Deal Protocol` indicate an intent to accept the deal.
+
+Execution now moves back to the `StorageClient`
+
+7. The `StorageClient` opens a push request using the `Data Transfer Module`, and sends the request to the provider along with a voucher containing the CID for the `StorageDealProposal` 
+8. The `StorageProvider` checks the voucher and verifies the CID matches an open proposal, and, if so, accepts the data transfer request from the `StorageClient`.
+9. The `Data Transfer Module` now transfers for the data to be stored from the  `StorageClient` to the `StorageProvider` using `GraphSync`.
+10. Once complete, the `Data Transfer Module` notifies the `StorageProvider`
+11. The `StorageProvider` recalculates the piece commitment (`CommP`) from the data that was transferred and verifies it matches the piece commitment in the the `StorageDealProposal`
+
+# Publishing
+
+Data is now transferred, both parties have agreed, and it's time for the `StorageProvider` to publish the deal.
```

Comment:
wait... whoa I gotta check on this -- I thought multiple discussions had arrived at the conclusion that the Storage Provider ALWAYS publishes. Otherwise there are multiple other issues we need to revisit -- such has not having a concept of a counter signed proposal. (rather the counter signature is simply the publish message -- which if the client can do it, is not sufficient, cause there is potentially no signature from the provider).

@zixuanzh did I misunderstand something?

Reply:
My apology for the late review on this PR, StorageProvider "always" publishes because the "counter signature" is a standard message signature by the provider. However, if StorageProvider decides to send this signed on chain message to the client before calling `PublishStorageDeal` then client can publish the deal on chain. Client's funds are not locked until the deal is published and a published deal that is not activated within some window will result in on chain penalty, cc @jzimmerman on the last sentence.